go.mod

```diff
@@ -2,10 +2,15 @@ module github.com/Worldcoin/hubble-commander
 
 go 1.17
 
+// fixes a bug where badger does not garbage collect if you are only updating a single
+// key over and over again
+replace github.com/dgraph-io/badger/v3 => github.com/lithp/badger/v3 v3.2103.2-hotfix
```

Comment:
What's the plan for this ? Are we making this change upstream ?

Reply:
There are two things which need to happen upstream. The first has already been merged but hasn't yet made it into a release: https://github.com/dgraph-io/badger/pull/1784

The second is a fix to that PR which fixes a deadlock, I'm talking about it with them in that link. I'm optimistic that within a few months upstream will cut a new release and we won't need to use a custom fork anymore.